### PR TITLE
MDEXP-276 Restrict creating a mapping profile to supported fields only

### PR DESCRIPTION
**Pull request by illia-borysenko**
https://issues.folio.org/browse/MDEXP-276

##Description
Add validation for transformations during creation and updating mapping profile, match transformation fieldId with existing transformationFields

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [x] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [x] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [x] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java [hunk 1]
```diff
@@ -105,4 +115,36 @@
     }
     return mappingProfileDao.delete(mappingProfileId, tenantId);
   }
+
+  @Override
+  public Future<Void> validate(MappingProfile mappingProfile, OkapiConnectionParams params) {
+    if (CollectionUtils.isNotEmpty(mappingProfile.getTransformations())) {
+       return transformationFieldsService.getTransformationFields(params)
+        .compose(transformationFieldCollection -> {
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java:123]: let's write this compose on 122 line

File: src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java [hunk 2]
```diff
@@ -105,4 +115,36 @@
...
+          List<TransformationField> transformationFields = transformationFieldCollection.getTransformationFields();
+          for (Transformations transformation : mappingProfile.getTransformations()) {
+            String fieldId = transformation.getFieldId();
+            if (StringUtils.isBlank(fieldId)) {
+              throw new ServiceException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY, "Field id is missing for mapping profile transformation");
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java:128]: can we append transformation.fieldId to the message

File: src/main/java/org/folio/rest/impl/DataExportImplMappingProfilesImpl.java [hunk 3]
```diff
@@ -1,28 +1,42 @@
...
 import org.folio.rest.jaxrs.resource.DataExportMappingProfiles;
 import org.folio.rest.tools.utils.TenantTool;
 import org.folio.service.profiles.mappingprofile.MappingProfileService;
+import org.folio.service.transformationfields.TransformationFieldsService;
 import org.folio.spring.SpringContextUtil;
 import org.folio.util.ExceptionToResponseMapper;
 import org.folio.util.OkapiConnectionParams;
 import org.springframework.beans.factory.annotation.Autowired;
 
+import javax.swing.text.html.Option;
```
Comment (Igor-Gorchakov) [inline src/main/java/org/folio/rest/impl/DataExportImplMappingProfilesImpl.java:28]: let's use java.util.Optional everywhere

**Review by Igor-Gorchakov — APPROVED** [review]

**Review by andrii-novytskyi — APPROVED** [review]
I think this ticket require API test, don`t forget to add it, and check unit tests, a  lot of failures there

File: src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java (hunk 2 above)
Comment (illia-borysenko) [reply src/main/java/org/folio/service/profiles/mappingprofile/MappingProfileServiceImpl.java:128]: In this case it's blank,  so we cannot

**Pull request merged**